### ISSUE TITLE
Stockfish doesn't compile anymore with Visual Studio

### ISSUE THREAD

**Issue by jackbond** [official-stockfish/Stockfish#3289]
I apologize if I'm asking something which has been answered in a FAQ, but I'm trying to compile Stockfish 12 in Visual Studio 2019, and I'm getting some compiler errors I don't know how to resolve. I downloaded the source from here, https://stockfishchess.org/download/ and added the source files to a Visual Studio project, but I'm getting some errors like this...

Does anyone have a solution file compiling the v12 code that they'd be willing to share? Thanks.

![image](https://user-images.githubusercontent.com/2799232/103936978-fba73a80-50dc-11eb-8513-a71589c849a7.png)

**Comment by BM123499** [comment]
@jackbond if you change the line 287 from `type.h` from `return Score((int)((unsigned int)eg << 16) + mg);` to `return Score((unsigned int)((unsigned int)eg << 16) + mg);` you may silent this error.

**Comment by vondele** [comment]
so, is that still an issue with current master, and if so is that an issue in Stockfish or a problem with of the compiler?

**Comment by JackWright347** [comment]
@jackbond you might want to use this solution https://github.com/official-stockfish/Stockfish/pull/3585

**Comment by vondele** [comment]
we recommend to use gcc / clang on windows and refer to the following wiki for the best approach:
https://github.com/glinscott/fishtest/wiki/Building-Stockfish-on-Windows

**Comment by mstembera** [comment]
What user friendly debugger do people use on Windows?    Nice debugging is the primary reason I use MSVC.

**Comment by vondele** [comment]
gdb.... for a somewhat unusual definition of 'user friendly'. It does have a number of GUI frontends that make it more mouse friendly https://sourceware.org/gdb/wiki/GDB%20Front%20Ends 
but yeah, that's for sure a point.

**Comment by soshimozi** [comment]
I'm trying to find some documentation on how to create a windows static library (or dll) from the Stockfish code base.  I was able to get it to build in MingGW, but I don't think AR knows about COFF and won't be able to generate DLL files.  I tried creating an SO with AR and that wasn't a problem.  I can't consume the SO in a windows application though.  I want to use the engine as a true engine without the CLI front end.  I also don't want to have to pipe - this is a Windows application.
